src/DataSource/DoctrineDataSource.php

```diff
@@ -365,10 +365,12 @@ public function addAggregationColumn($aggregation_type, $column)
 	 */
 	public function getAggregationData()
 	{
+		if (count($this->aggregations) === 0) {
+			return [];
+		}
 		$this->data_source->resetDQLPart('select');
-		$root_alias = $this->data_source->getRootAliases()[0];
 		foreach ($this->aggregations as $column => $aggregation_type) {
-			$query = $this->data_source->select("$aggregation_type($root_alias.$column) AS $column");
+			$query = $this->data_source->select("$aggregation_type(" . $this->checkAliases($column) . ") AS $column");
```

Comment:
Just looking at it now, you should probably use `addSelect` instead and be careful with `... AS $column` as it can contain string looking like this: `p.name`. You should sanitize it.